mlir/tools/mlir-miopen-driver/mlir-miopen-driver.cpp

```diff
@@ -2430,6 +2509,47 @@ int main(int argc, char **argv) {
     }
   }
 
+  // Populate the module for gpu validation.
+  SmallVector<std::string, 4> kernels_v;
+  if (populateValidationWithGPU.getValue() &&
+      (xdlopsV2.getValue() || dataType != builder.getF32Type())) {
+    // use non-xdlops kernels to verify xdlops kernels
+    if (xdlopsV2.getValue())
```

Comment:
So, in other words, we always use non-xdlops kernels for verification? This feels like the logic should be simpler, unless we're using xdlops to verify non-xdlops